x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -79,15 +87,14 @@ export function isSourceDataChartableForDetector(job, detectorIndex) {
     // If the datafeed uses script fields, we can only plot the time series if
     // model plot is enabled. Without model plot it will be very difficult or impossible
     // to invert to a reverse search of the underlying metric data.
-    const usesScriptFields = _.has(job, 'datafeed_config.script_fields');
-    if (isSourceDataChartable === true && usesScriptFields === true) {
+    if (isSourceDataChartable === true && typeof job.datafeed_config?.script_fields === 'object') {
       // Perform extra check to see if the detector is using a scripted field.
-      const scriptFields = usesScriptFields ? _.keys(job.datafeed_config.script_fields) : [];
+      const scriptFields = Object.keys(job.datafeed_config.script_fields);
       isSourceDataChartable =
-        scriptFields.indexOf(dtr.field_name) === -1 &&
-        scriptFields.indexOf(dtr.partition_field_name) === -1 &&
-        scriptFields.indexOf(dtr.by_field_name) === -1 &&
-        scriptFields.indexOf(dtr.over_field_name) === -1;
+        scriptFields.indexOf(String(dtr.field_name)) === -1 &&
+        scriptFields.indexOf(String(dtr.partition_field_name)) === -1 &&
+        scriptFields.indexOf(String(dtr.by_field_name)) === -1 &&
+        scriptFields.indexOf(String(dtr.over_field_name)) === -1;
```

Comment:
if one of these is `undefined` e.g. `field_name`, converting it to a string with `String(dtr.field_name)` will produce `'undefined'`.
there would be a risk that `'undefined'` is a real name of a field. albeit a strange one.

Reply:
thanks for spotting this, was a temp change, forgot to remove